caikit/runtime/utils/servicer_util.py

```diff
@@ -403,7 +403,11 @@ def build_caikit_library_request_dict(
                 # Remove empty iterables since we cannot distinguish between
                 # unset and empty repeated fields
                 field_value = getattr(request, field.name)
-                if isinstance(field_value, Iterable) and len(field_value) == 0:
+                if (
+                    isinstance(field_value, Iterable)
+                    and len(field_value) == 0
+                    and not isinstance(field_value, (str, bytes))
```

Comment:
Super NIT, but it would probably be slightly more efficient to evaluate the `not isinstance` first